internal/globalconfig/globalconfig.go

```diff
@@ -0,0 +1,28 @@
+// Package globalconfig stores configuration which applies globally to both the tracer
+// and integrations.
+package globalconfig
+
+import "sync"
+
+var cfg = &config{}
+
+type config struct {
+	mu            sync.RWMutex
+	analyticsRate float64
+}
+
+// AnalyticsRate returns the sampling rate at which events should be marked. It uses
+// synchronizing mechanisms, meaning that for optimal performance it's best to read it
```

Comment:
Is there a particular performance concern to make this comment necessary?
The defer is probably more "expensive" than the lock itself.

Reply:
I think it's necessary for contributors which might use this API. It's easy to just try and get away with reading it every time. This makes it clear that it should be read once.

Reply:
I mean there's no _real_ performance concern but I thought it'd help. Do you think we should remove it?